design/fake-client.rst

```diff
@@ -0,0 +1,153 @@
+.. Copyright 2016 IBM Corp. All Rights Reserved.
+..
+.. Licensed under the Apache License, Version 2.0 (the "License");
+.. you may not use this file except in compliance with the License.
+.. You may obtain a copy of the License at
+..
+..    http://www.apache.org/licenses/LICENSE-2.0
+..
+.. Unless required by applicable law or agreed to in writing, software
+.. distributed under the License is distributed on an "AS IS" BASIS,
+.. WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+.. See the License for the specific language governing permissions and
+.. limitations under the License.
+..
+
+Design for fake client
+======================
+
+This document describes requirements and possible design options for a fake
+zhmcclient.
+
+Requirements
+------------
+
+* Must avoid the need for a real HMC and still allow usage of the zhmcclient
+  API.
+* Both read-only and read-write operations must be supported.
+* Read-write operations must reflect their changes properly (i.e. they must be
+  visible in corresponding read operations).
+* The resource state of the faked HMC must be preloaded (e.g. for resources
+  that cannot be created by clients), and its resource descriptions must be
+  easy to use (i.e. not in terms of REST URIs.
+* It must be possible to define testcases with error injection.
+
```

Comment:
Although it is probably implied, it would be great to have here the explicit statement that the solution, however it might look like, must be usable in *unit tests* of library users. This means the solution needs to perform fast and doesn't rely on external things, like a small HTTP server or similar.